autocodesign/profiles.go

```diff
@@ -199,14 +201,43 @@ func (m profileManager) ensureBundleID(bundleIDIdentifier string, entitlements E
 	}
 
 	if err := m.client.SyncBundleID(*bundleID, entitlements); err != nil {
-		return nil, fmt.Errorf("failed to update bundle ID capabilities: %s", err)
+		return nil, fmt.Errorf("failed to update bundle ID capabilities: %w", err)
 	}
 
 	m.bundleIDByBundleIDIdentifer[bundleIDIdentifier] = bundleID
 
 	return bundleID, nil
 }
 
+func (m profileManager) ensureProfileWithRetry(profileType appstoreconnect.ProfileType, bundleIDIdentifier string, entitlements Entitlements, certIDs, deviceIDs []string, minProfileDaysValid int) (*Profile, error) {
+	var profile *Profile
+	// Accessing the same Apple Developer Portal team can cause race conditions (parallel CI runs for example).
+	// Between the time of finding and downloading a profile, it could have been deleted for example.
+	if err := retry.Times(5).Wait(10 * time.Second).TryWithAbort(func(attempt uint) (error, bool) {
```

Comment:
Should we maybe switch up the numbers here?

Right now it does a low amount of retries with a large waiting period between them. If we do `Times(10).Wait(5 * time.Second)` then we could waiting times at least in half which means we could gain a few seconds for the user. 

You mentioned earlier that you saw that on some occasions on the profile was not empty on the third try and also wrote in the comment that this also comes up when there are multiple parallel accesses to the portal. So I am not sure in how long tasks take on the portal but let me throw in the following idea and hear your opinion about it.

What if we do 
```
if err := retry.Times(10).Wait(3 * time.Second).TryWithAbort(func(attempt uint) (error, bool) {
```

Reply:
I would go with 10 seconds as the retry only happens if there is someone else editing the Developer Portal. So waiting a bit should help so they can finish whatever they are doing. I would consider an exponential backoff (3 -> 6 -> 12) also.